### PR TITLE
Robust vectors

### PR DESCRIPTION
**Pull request by madrugado**

### CODE REVIEW

File: deeppavlov/datasets/robust_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        self.seq_length = seq_length
+
+        self.chars = chars
+        self.vocab = vocab
+        self.noise_level = 0.0
+        vocab_file = os.path.join(data_dir, "vocab.pkl")
+        tensor_file = os.path.join(data_dir, "data.npy")
+        letter_vocab_file = os.path.join(data_dir, "letter_vocab.npy")
+
+        if not (os.path.exists(vocab_file)
```
Comment (my-master) [inline deeppavlov/datasets/robust_dataloader.py]: 1. pathlib is used in the project

File: deeppavlov/datasets/robust_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            cPickle.dump(self.tensor, f)
+        np.save(letter_vocab_file, self.letter_vocab)
+
+    def load_preprocessed(self, vocab_file, tensor_file, letter_vocab_file):
+        with codecs.open(vocab_file, 'rb') as f:
+            self.chars = cPickle.load(f)
+        self.vocab_size = len(self.chars) + 1
+        self.vocab = dict(zip(self.chars, range(1, len(self.chars) + 1)))
+        self.vocab[""] = 0
+        with codecs.open(tensor_file, "rb") as f:
```
Comment (my-master) [inline deeppavlov/datasets/robust_dataloader.py]: 2. The function should be imported from deeppavlov.common.file

File: deeppavlov/datasets/robust_dataloader.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+from glob import glob
+
+from nltk import sent_tokenize, word_tokenize
+from six.moves import cPickle
+import numpy as np
+from tqdm import tqdm
+from collections import Counter
+
+
+class TextLoader:
```
Comment (my-master) [inline deeppavlov/datasets/robust_dataloader.py]: 3. Add a docstring.

Comment (my-master) [reply deeppavlov/datasets/robust_dataloader.py]: 4. The class is not wrapped into DeepPavlov abstractions. Looks like this class should be inherited from `DatasetReader` class and should have a proper name (sth like "BlaBlaReader").
Also there is a separate base class `DefaultVocabulary` which should be used here. If `DefaultVocabulary` is not good for this usecase, a separate Vocabulary class should be written.

File: deeppavlov/models/embedders/rove_embedder.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    @check_attr_true('train_now')
+    def train(self, data_loader, **kwargs):
+        print(':: creating new  model\n')
+        self._train_step(data_loader)
+        self.save()
+
+    def infer(self, sentence: str, *args, **kwargs):
+        return self._encode(sentence)
+
+    def load(self):
```
Comment (my-master) [inline deeppavlov/models/embedders/rove_embedder.py]: 5. Why `TFModel.save()` and `TFModel.load()` are not used here? If they are no good for this usecase, this is a serious lack of flexibility in the abstract class and the abstract class methods should be rewritten.

File: requirements.txt [hunk 5]
```diff
@@ -2,7 +2,6 @@ lxml==4.1.1
 tqdm==4.19.4
 requests==2.18.4
 gensim==2.3.0
-tensorflow_gpu==1.4.0
```
Comment (my-master) [inline requirements.txt]: 6. Why?

**Comment by my-master** [comment]
Please read about [git semantic commits](https://seesparkbox.com/foundry/semantic_commit_messages) and make proper commit messages.

File: requirements.txt (hunk 5 above)
Comment (madrugado) [reply requirements.txt]: to run the library on machines without GPU

File: deeppavlov/datasets/robust_dataloader.py (hunk 2 above)
Comment (madrugado) [reply deeppavlov/datasets/robust_dataloader.py]: which one?

Comment (my-master) [reply deeppavlov/datasets/robust_dataloader.py]: save_cpickle, load_cpickle - make these functions in deeppavlov.common.file and import from there

File: requirements.txt (hunk 5 above)
Comment (my-master) [reply requirements.txt]: Ok. Anyway, I don't know who inserted this req. : )

Comment (my-master) [reply requirements.txt]: Should be solved more cleverly in the future.

**Pull request closed without merging**